narwhals/_compliant/expr.py

```diff
@@ -349,7 +349,7 @@ def __init__(
         implementation: Implementation,
         backend_version: tuple[int, ...],
         version: Version,
-        call_kwargs: dict[str, Any] | None = None,
+        scalar_kwargs: dict[str, Any] | None = None,
```

Comment:
Should this be typed more strictly now?

```suggestion
        scalar_kwargs: dict[str, PythonLiteral] | None = None,
```

Apologies if I've misunderstood the name change 😅

Reply:
yup thanks

it requires a handful of casts though, right?

Reply:
> it requires a handful of casts though, right?

Yeah I think your options are either that or a `TypedDict` that looks *something like* this:

```py
from __future__ import annotations

from typing import TYPE_CHECKING
from typing import TypedDict

if TYPE_CHECKING:
    from narwhals.typing import RankMethod

class _ScalarKwargs(TypedDict, total=False):
    ddof: int
    n: int
    reverse: bool
    method: RankMethod
    descending: bool
```

You'd still be checking for the presence of keys, but when you find them - they'll be the right type

Reply:
oooooh I'd forgotten about `total=False` - thanks! 😍 that's real nice

Reply:
aha I see you started down that route yesterday (https://github.com/narwhals-dev/narwhals/pull/2555/commits/cba568dbe1aaf19f1d8d99697b5b898bc914a500)